### PR TITLE
mgr/rook: make `ceph orch device ls` use info from LSO

### PR DESCRIPTION
**Pull request by josephsawaya**
Added the CustomObjectsApi to the Rook orchestrator module so it can fetch custom resources from the cluster. 

Patched the module to fetch it's information on devices from the LSO's `LocalVolumeDiscoveryResult` CRs. 

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/rook/rook_cluster.py [hunk 1]
```diff
@@ -191,16 +189,20 @@ def _watch(self, res_ver: Optional[str]) -> None:
 class RookCluster(object):
     # import of client.CoreV1Api must be optional at import time.
     # Instead allow mgr/rook to be imported anyway.
-    def __init__(self, coreV1_api: 'client.CoreV1Api', batchV1_api: 'client.BatchV1Api', rook_env: 'RookEnv'):
+    def __init__(self, coreV1_api: 'client.CoreV1Api', batchV1_api: 'client.BatchV1Api', customObjects_api: 'client.CustomObjectsApi', rook_env: 'RookEnv'):
         self.rook_env = rook_env  # type: RookEnv
         self.coreV1_api = coreV1_api  # client.CoreV1Api
+        self.customObjects_api =customObjects_api # type: client.CustomObjectsApi
```
Comment (liewegas) [inline src/pybind/mgr/rook/rook_cluster.py:195]: whitespace

File: src/pybind/mgr/rook/rook_cluster.py [hunk 2]
```diff
@@ -191,16 +189,20 @@ def _watch(self, res_ver: Optional[str]) -> None:
...
         #  TODO: replace direct k8s calls with Rook API calls
         # when they're implemented
-        self.inventory_maps: KubernetesResource[client.V1ConfigMapList] = KubernetesResource(self.coreV1_api.list_namespaced_config_map,
-                                                 namespace=self.rook_env.operator_namespace,
-                                                 label_selector="app=rook-discover")
+        
+        self.inventory_maps = KubernetesResource(self.customObjects_api.list_cluster_custom_object,
+                                                 group="local.storage.openshift.io",
+                                                 version="v1alpha1",
+                                                 plural="localvolumediscoveryresults")
```
Comment (sebastian-philipp) [inline src/pybind/mgr/rook/rook_cluster.py:204]:
```
rook/rook_cluster.py: note: In member "__init__" of class "RookCluster":
rook/rook_cluster.py:201: error: Need type annotation for "inventory_maps"
```

File: src/pybind/mgr/rook/rook_cluster.py [hunk 3]
```diff
@@ -241,9 +243,9 @@ def rook_api_post(self, path: str, **kwargs: Any) -> Any:
         return self.rook_api_call("POST", path, **kwargs)
 
     def get_discovered_devices(self, nodenames: Optional[List[str]] = None) -> Dict[str, dict]:
-        def predicate(item: client.V1ConfigMapList) -> bool:
+        def predicate(item) -> bool:
```
Comment (sebastian-philipp) [inline src/pybind/mgr/rook/rook_cluster.py:246]:
```
rook/rook_cluster.py: note: In function "get_discovered_devices":
rook/rook_cluster.py:246: error: Function is missing a type annotation for one or more arguments
```

File: src/pybind/mgr/rook/module.py [hunk 4]
```diff
@@ -179,18 +182,14 @@ def get_inventory(self, host_filter: Optional[orchestrator.InventoryFilter] = No
...
-                        available = False,
-                        rejected_reasons=['device data coming from ceph-volume not provided'],
-                    ))
+                devs.append(inventory.Device(
+                    path = d['path'],
+                    sys_api = dict(
+                        rotational = '1' if d['property']=='Rotational' else '0',
+                        size = d['size']
+                    ),
+                    available = True if d['status']['state']=='Available' else False,
```
Comment (jmolmo) [inline src/pybind/mgr/rook/module.py:191]: available = d['status']['state']=='Available'

**Pull request closed without merging**